### ISSUE TITLE
Using direct parent/child relationship instead of slots in Tab components

### ISSUE THREAD

**Issue by muratcorlu** [Trendyol/baklava#517]
We have some child components those intended to work only inside another Baklava component. Some examples are:

* `bl-select-option` inside `bl-select`
* `bl-radio` inside `bl-radio-group`
* `bl-dropdown-item` -optionally- inside `bl-dropdown-group` inside `bl-dropdown`

And for those elements we don’t use `slot`s in parent elements to place child elements in related place. But `Tab` is the exception here. 

```html
<bl-tab-group>
  <bl-tab slot=“tabs” name=“main”>Overview<bl-tab>
  …
</bl-tab-group>
```

But I think to have consistent and more intuitional approach, we would drop the requirement for using `slot` attribute for `bl-tab` element like this:

```html
<bl-tab-group>
  <bl-tab name=“main”>Overview<bl-tab>
  <bl-tab-panel tab=“main”> Tab content </bl-tab-panel>
</bl-tab-group>
```

Currently when we write like this, layout is being broken without any errors. I thing we can just make this smarter without a breaking change by checking if `slot` attribute is provided or not, and if not, `bl-tab` can register itself to `bl-tab-group` (as we did in other examples above) and that registration can place it to proper place in parent element accordingly.

**Comment by mehmetranas** [comment]
I agree with you, but do you have any suggestions about how to implement it? Because we use two slots in a `bl-tab-group`. Injecting `bl-tab` and `bl-tab-panel` without using a slot prop will be a difficult challenge.
